app/src/androidTest/java/com/mapbox/maps/testapp/viewport/ViewportPluginTest.kt

```diff
@@ -70,52 +68,135 @@ class ViewportPluginTest : BaseMapTest() {
 
   @Test
   fun transitionToDefaultTransition() {
-    assumeTrue(
-      "Can only run on API Level 24 or newer because of difference of animator behaviour.",
-      Build.VERSION.SDK_INT > 23
-    )
+    val latch = CountDownLatch(1)
     handler.post {
-      viewportPlugin.transitionTo(followPuckViewportState)
       assertEquals(0.0, mapView.getMapboxMap().cameraState.bearing, EPS)
       mapView.getMapboxMap().cameraState.center.assertEquals(NULL_ISLAND)
-      locationProvider.locationConsumers.forEach { it.onLocationUpdated(TEST_POINT) }
-      locationProvider.locationConsumers.forEach { it.onBearingUpdated(TEST_BEARING) }
+      // immediate update location puck to test location.
+      locationProvider.locationConsumers.forEach {
+        it.onLocationUpdated(
+          TEST_POINT,
+          options = { duration = 0 }
+        )
+      }
+      locationProvider.locationConsumers.forEach {
+        it.onBearingUpdated(
+          TEST_BEARING,
+          options = { duration = 0 }
+        )
+      }
+      // transition to the followPuckViewportState with default transition
+      viewportPlugin.transitionTo(followPuckViewportState) {
+        latch.countDown()
+      }
     }
 
-    val latch = CountDownLatch(1)
-    latch.await(4000, TimeUnit.MILLISECONDS)
+    // Wait for 5 seconds since the default transition time is 3.5 seconds
```

Comment:
Is it still 3.5 seconds for the immediate transition?

Reply:
No, for immediate transition we use 2 seconds timeout as below, and this is just worst case timeout, we do count down the latch in the transition finished callback, so it should Finnish pretty fast.

Reply:
actually updated to 200ms timeout, as I don't see this operation should take that much time, if it does, then probably there's something wrong.

Reply:
It's still 5 seconds delay here, seems 200 ms was only changed in a differet test

Reply:
yeah, here it should be 5 second delay in this test, as the transition itself takes 3.5 seconds, we need to wait for that transition to finish.